### ISSUE TITLE
Make retries for finding build customizable

### ISSUE THREAD

**Issue by nilsreichardt** [codemagic-ci-cd/cli-tools#350]
https://github.com/codemagic-ci-cd/cli-tools/blob/e1aab8612f57b618a6476beeee5509576bce77ac/src/codemagic/tools/_app_store_connect/actions/publish_action.py#L411

We have sometimes the issue that finding the build times out, like here:

* https://github.com/SharezoneApp/sharezone-app/actions/runs/6216197154/job/16869845823

```
Build has finished uploading but is processing on App Store Connect side. Could not find the build matching the uploaded version yet. Waiting 30 seconds to try again, 1 attempts remaining.
The build was successfully uploaded to App Store Connect but processing the corresponding artifact "build/ios/ipa/sharezone.ipa" by Apple took longer than expected. Further actions like updating the What to test information or submitting the build to beta review could not be performed at the moment but can be completed manually in TestFlight once the build has finished processing.
Failed to publish build/ios/ipa/sharezone.ipa
```

I would appreciate it if you could customize this with a parameter, like `--max-finding-build-retries`.

**Comment by nilsreichardt** [comment]
Thanks, @priitlatt <3

**Comment by priitlatt** [comment]
Forgot to mention it here that this is released now and available in [version 0.46.0](https://github.com/codemagic-ci-cd/cli-tools/releases/tag/v0.46.0).
